packages/viewer/src/rendering/canvas.ts

```diff
@@ -46,7 +46,11 @@ function reportTimings(api: StreamApi, meter: TimingMeter): void {
     .map(t => ({ receiveToPaintDuration: toProtoDuration(t.duration) }));
 
   if (timings.length > 0) {
-    api.recordPerformance({ timings: timings }, false);
+    try {
+      api.recordPerformance({ timings: timings }, false);
+    } catch (e) {
+      console.warn('Unable to record performance: ', e);
+    }
```

Comment:
is there something specific about this API call that is prone to errors? or the try/catch because it's not an essential function?

Reply:
No, this is explained above. The request is sent on the websocket. I don't think that we should be indicating these errors to the client, but still have these tracked and logged as warnings. I think a better solution here is to not record performances on the fss/ws stream and potentially post these metrics to a restful endpoint.